src/layouts/InterfaceLayout/containers/InteractWithContractContainer/InteractWithContractContainer.vue

```diff
@@ -1,84 +1,349 @@
+    'currency-picker': CurrencyPicker,
+    'confirm-modal': ConfirmModal,
+    'success-modal': SuccessModal
   },
   data () {
     return {
-      existingContracts: [
-        {
-          name: 'Battle Of Thermopy wefweoifjwfo ewrofijweo',
-          value: '1'
-        },
-        {
-          name: 'Battle Of Thermopy wefweoifjwfo ewrofijweo g',
-          value: '2'
-        },
-        {
-          name: 'Battle Of Thermopy wefweoifjwfo ewrofijweo gf',
-          value: '3'
-        },
-        {
-          name: 'Battle Of Thermopy wefweoifjwfo ewrofijw',
-          value: '4'
+      abi: '',
+      address: '',
+      interact: false,
+      validAbi: false,
+      validAddress: false,
+      methods: [],
+      selectedMethod: {},
+      result: '',
+      writeInputs: {},
+      inputsFilled: false,
+      loading: false,
+      resType: 'string',
+      gasLimit: 21000,
+      value: 0,
+      data: '',
+      raw: {},
+      nonce: 0,
+      signedTx: '',
+      transactionFee: 0
+    }
+  },
+  methods: {
+    selectedCurrency (currency) {
+      if (currency.abi === '') {
+        this.abi = ''
+      } else {
+        this.abi = JSON.stringify(currency.abi)
+      }
+      this.address = currency.address
+    },
+    checkType (type) {
+      if (type === 'address' || type === 'string' || type === 'byte') {
+        return 'string'
+      } else {
+        return 'number'
+      }
+    },
+    selectFunction (method) {
+      const contract = new this.$store.state.web3.eth.Contract(
+        [method],
+        this.address
+      )
+      if (method.constant === true && method.inputs.length === 0) {
+        contract.methods[method.name]()
+          .call({ from: this.$store.state.wallet.getAddressString() })
+          .then(res => {
+            this.result = res
+          })
+          .catch(err => {
+            console.log(err)
+          })
+      } else {
+        this.result = ''
+      }
+      this.selectedMethod = method
+    },
+    copyToClipboard (ref) {
+      this.$refs[ref].select()
+      document.execCommand('copy')
+    },
+    deleteInput (ref) {
+      this.$refs[ref].value = ''
+    },
+    switchView (direction) {
+      switch (direction) {
+        case 'forward':
+          this.methods = JSON.parse(this.abi)
+            .map(func => func)
+            .filter(func => func.type !== 'constructor')
+            .filter(func => func.constant !== undefined)
+          this.interact = true
+          break
+        default:
+          this.interact = false
+      }
+    },
+    async write () {
+      const contract = new this.$store.state.web3.eth.Contract([this.selectedMethod], this.address)
+      const params = Object.keys(this.writeInputs).map(input => this.writeInputs[input])
+      this.loading = true
+      if (this.selectedMethod.constant === true) {
+        contract.methods[this.selectedMethod.name](...params)
+          .call({ from: this.$store.state.wallet.getAddressString() })
```

Comment:
to answer your question
```
var MyContract = new web3.eth.contract(abi, address);
MyContract.methods.myFunction().call()
.then(console.log);
> Result {
    myNumber: '23456',
    myString: 'Hello!%',
    0: '23456', // these are here as fallbacks if the name is not know or given
    1: 'Hello!%'
}
```
if it is a single output, then `call` will just return that, but if it is multiple outputs itll return both an object or an array

> The return value(s) of the smart contract method. If it returns a single value, it’s returned as is. If it has multiple return values they are returned as an object with properties and indices:

Reply:
I just end up looping through the object keys and using that to render the results instead.